File: experiments/006-SmalltalkVsJava/cells/java-gradle_1-Empty_free/2022-1c-recuperatorio-parcial1/20260915-110924-39025/analysis/acceptance-project/src/main/java/customerimporter/OldZipCode.java
package customerimporter;

import java.util.function.Supplier;

public class OldZipCode extends ZipCode {

    // instance creation

    static boolean appliesTo(String aZipCode) {
        return Character.isDigit(aZipCode.charAt(0));
    }

    // error messages

    public static String invalidOldZipCodeErrorDescription() {
        return "Invalid old zipcode";
    }

    // initialization

    OldZipCode(String aCode) {
        super(aCode);
    }

    // zip code

    public Object value() {
        return number();
    }

    public int number() {
        return Integer.parseInt(code());
    }

    // type

    public boolean isOld() {
        return true;
    }

    public Object oldZipCodeIfNone(Supplier<Object> aNoneBlock) {
        return value();
    }

    // validation

    protected void assertIsValid() {
        assertThat(Characters.areAllDigits(code()));
        assertThat(number() >= 1000 && number() <= 9999);
    }

    protected String invalidZipCodeErrorDescription() {
        return invalidOldZipCodeErrorDescription();
    }
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2022-1c-recuperatorio-parcial1/20260921-213435-24827/workdir/src/customerimporter/old_zip_code.py
from customerimporter.zip_code import ZipCode


class OldZipCode(ZipCode):

    # instance creation

    @classmethod
    def can_represent(cls, a_zip_code):
        return a_zip_code[0].isdigit()

    # error messages

    @classmethod
    def invalid_value_error_description(cls):
        return "Invalid old zipcode"

    # validation

    def assert_is_valid(self):
        self._assert_that(self._are_all_digits(self._value))
        self._assert_that(self.code() >= 1000 and self.code() <= 9999)

    # zip code

    def code(self):
        return int(self._value)

    # old zip code

    def has_old_zip_code(self):
        return True

    def old_zip_code_if_none(self, a_none_block):
        return self.code()
